src/lib/ddev/init_projecttype.tsx

```diff
@@ -0,0 +1,145 @@
+ */
+const knownDDEVProjectTypes = [
+  "backdrop",
+  "craftcms",
+  "django4",
+  "drupal6",
+  "drupal7",
+  "drupal",
+  "laravel",
+  "magento",
+  "magento2",
+  "php",
+  "python",
+  "shopware6",
+  "silverstripe",
+  "typo3",
+  "wordpress",
+];
+
+async function getAppVersion(
+  client: MittwaldAPIV2Client,
+  appId: string,
+  appVersionId: string,
+): Promise<AppVersion> {
+  const r = await client.app.getAppversion({
+    appId,
+    appVersionId,
+  });
+
+  assertStatus(r, 200);
+  return r.data;
+}
+
+/**
+ * Determines the DDEV project type to use for the given app installation.
+ *
+ * This is done according to the following rules:
+ *
+ * 1. If an explicit override was specified (typically using the --override-type
+ *    flag), use that.
+ * 2. If the app installation is a known app type (e.g. TYPO3, WordPress, etc.),
+ *    use the corresponding DDEV project type.
+ * 3. Prompt the user to interactively select a DDEV project type.
+ */
+export async function determineProjectType(
+  r: ProcessRenderer,
+  client: MittwaldAPIV2Client,
+  inst: AppInstallation,
+  typeOverride: string,
+): Promise<string> {
+  if (typeOverride !== "auto") {
+    r.addInfo(<ProjectTypeInfoOverride type={typeOverride} />);
+    return typeOverride;
+  }
+
+  const determinedProjectType = await determineProjectTypeFromAppInstallation(
+    client,
+    inst,
+  );
+  if (determinedProjectType !== null) {
+    r.addInfo(<ProjectTypeInfoAuto type={determinedProjectType} />);
+    return determinedProjectType;
+  }
+
+  return await promptProjectTypeFromUser(r);
+}
+
+/**
+ * Interactively prompts the user to select a DDEV project type.
+ *
+ * The list of known project types is hardcoded in this module.
+ */
+async function promptProjectTypeFromUser(r: ProcessRenderer): Promise<string> {
+  return r.addSelect(
+    "select the DDEV project type",
+    knownDDEVProjectTypes.map((t) => ({ value: t, label: t })),
+  );
+}
+
+/**
+ * Determines the project type to use for the given app installation.
+ *
+ * In most cases, this is a simple mapping from known app IDs to DDEV project
+ * types. In some cases, we might need to know the specific (major) version of
+ * an app to determine the correct project type (for example, DDEV has separate
+ * project types for the different Drupal major versions).
+ */
+export async function determineProjectTypeFromAppInstallation(
+  client: MittwaldAPIV2Client,
+  inst: AppInstallation,
+): Promise<string | null> {
+  switch (inst.appId) {
+    case typo3Installer.appId:
+      return "typo3";
+    case wordpressInstaller.appId:
+      return "wordpress";
+    case shopware6Installer.appId:
+      return "shopware6";
+    case drupalInstaller.appId: {
+      const version = await getAppVersion(
+        client,
+        inst.appId,
+        inst.appVersion.desired,
+      );
+
+      const [major] = version.externalVersion.split(".");
+      return `drupal${major}`;
```

Comment:
Does this really work?

Available ddev project types (defined in line 22–24): `"drupal6", "drupal7", "drupal",`.  In the mStudio GUI it’s currently possible to install version `10.2.5`. This would result in `project-type=drupal10`, wouldn’t it?

Reply:
Ouch, good catch. 😖   I have to admit that I was completely oblivious as to which Drupal versions actually exist out there. Should've done my homework better... 😕 

Hang on...

Reply:
Fixed in 8f96ac6